### PR TITLE
[AIRFLOW-2640] Add Cassandra table sensor

### PR DESCRIPTION
**Pull request by sekikn**
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-2640
    - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a JIRA issue.

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:

Just like a partition sensor for Hive,
this PR adds a sensor that waits for
a table to be created in Cassandra cluster.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

tests.contrib.hooks.test_cassandra_hook:CassandraHookTest.test_table_exists
tests.contrib.sensors.test_cassandra_sensor:TestCassandraTableSensor

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### Documentation
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality
- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: airflow/contrib/sensors/cassandra_record_sensor.py [hunk 1]
```diff
@@ -29,9 +29,9 @@ class CassandraRecordSensor(BaseSensorOperator):
     primary keys 'p1' and 'p2' to be populated in keyspace 'k' and table 't',
     instantiate it as follows:
 
-    >>> CassandraRecordSensor(table="k.t", keys={"p1": "v1", "p2": "v2"},
-    ...     cassandra_conn_id="cassandra_default", task_id="cassandra_sensor")
-    <Task(CassandraRecordSensor): cassandra_sensor>
+    >>> t = CassandraRecordSensor(table="k.t", keys={"p1": "v1", "p2": "v2"},
```
Comment (Fokko) [inline airflow/contrib/sensors/cassandra_record_sensor.py]: Please give a more meaningful name

File: airflow/contrib/hooks/cassandra_hook.py [hunk 2]
```diff
@@ -158,6 +158,21 @@ def get_lb_policy(policy_name, policy_args):
...
+        Checks if a table exists in Cassandra
+
+        :param table: Target Cassandra table.
+                      Use dot notation to target a specific keyspace.
+        :type table: string
+        """
+        keyspace = None
+        if '.' in table:
+            keyspace, table = table.split('.', 1)
+        m = self.get_conn().cluster.metadata
```
Comment (Fokko) [inline airflow/contrib/hooks/cassandra_hook.py]: Please give a more meaningful name

File: airflow/contrib/hooks/cassandra_hook.py [hunk 3]
```diff
@@ -158,6 +158,21 @@ def get_lb_policy(policy_name, policy_args):
...
+        return (keyspace in m.keyspaces and
```
Comment (Fokko) [inline airflow/contrib/hooks/cassandra_hook.py]: If the keyspace is `None`, then this predicate will always evaluate to `False`, right?

**Review by Fokko — CHANGES_REQUESTED** [review]

File: airflow/contrib/hooks/cassandra_hook.py (hunk 3 above)
Comment (sekikn) [reply airflow/contrib/hooks/cassandra_hook.py]: Yeah, that's a mistake. `self.keyspace` should be used if a keyspace is not specified in CQL.

**Comment by sekikn** [comment]
Thanks for the review @Fokko! Updated the PR.

- fixed the bug Fokko found
- added tests which covers the above case
- assigned meaningful names to some variables

**Review by Fokko — APPROVED** [review]
LGTM, nice one

**Pull request closed without merging**